google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java

```diff
@@ -271,7 +272,38 @@ private boolean runningOnAppEngine() {
       return true;
     }
 
-    return false;
+    Class<?> systemPropertyClass = null;
+    try {
+      systemPropertyClass = forName("com.google.appengine.api.utils.SystemProperty");
```

Comment:
This line used to have what looked like a circular dependency problem but this looks better.